packages/dds/tree/src/test/edit-manager/editManger.spec.ts

```diff
@@ -0,0 +1,606 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { fail, strict as assert } from "assert";
+import { ChangeFamily } from "../../change-family";
+import { SeqNumber } from "../../changeset";
+import { Commit, EditManager } from "../../edit-manager";
+import { ChangeRebaser } from "../../rebase";
+import { AnchorSet } from "../../tree";
+import { makeArray, RecursiveReadonly } from "../../util";
+
+interface TestChangeset {
+    inputContext?: number;
+    outputContext?: number;
+    intentions: number[];
+}
+
+interface NonEmptyTestChangeset extends TestChangeset {
+    inputContext: number;
+    outputContext: number;
+}
+
+function isNonEmptyChange(change: RecursiveReadonly<TestChangeset>):
+	change is RecursiveReadonly<NonEmptyTestChangeset> {
+    return change.inputContext !== undefined && change.outputContext !== undefined;
+}
+
+class TestChangeRebaser implements ChangeRebaser<TestChangeset, TestChangeset, TestChangeset> {
+    private contextCounter: number = 0;
+    private intentionCounter: number = 0;
+
+    public compose(...changes: TestChangeset[]): TestChangeset {
+        let inputContext: number | undefined;
+        let outputContext: number | undefined;
+        const intentions: number[] = [];
+        for (const change of changes) {
+            inputContext ??= change.inputContext;
+            outputContext = change.outputContext ?? outputContext;
+            intentions.push(...change.intentions);
+        }
+        return {
+            inputContext,
+            outputContext,
+            intentions,
+        };
+    }
+
+    public invert(change: TestChangeset): TestChangeset {
+        return {
+            inputContext: change.outputContext,
+            outputContext: change.inputContext,
+            intentions: change.intentions.map((i) => -i),
```

Comment:
It is not meaningful which makes me think maybe we want it to be a set instead. What do you think?
If we keep it as an array then I think it would indeed be nice to reverse the order, just for inspectability's sake.

Reply:
Reversed them.